pkg/server/registry/apigroups/admin/computeclass/validator.go

```diff
@@ -39,6 +40,13 @@ func (s *ProjectValidator) Validate(ctx context.Context, obj runtime.Object) (re
 		}
 	}
 
+	// If a PriorityClass is specified, validate it exists
+	if cc.PriorityClassName != "" {
+		if err := s.client.Get(ctx, kclient.ObjectKey{Name: cc.PriorityClassName, Namespace: ""}, &schedulingv1.PriorityClass{}); err != nil {
+			return append(result, field.Invalid(field.NewPath("spec", "priorityClassName"), cc.PriorityClassName, err.Error()))
+		}
+	}
```

Comment:
Do we want to validate this on create? There are other instances where we don't validate such things. For example, we don't validate that a storage class exists when a volume class is created.

Reply:
Yeah so I had a similar thought. My thought process went slightly different so please let me know what you think.

If the PriorityClass isn't validated here, it is validated on App creation (like memory, cpu, etc is now). What felt different about that situation is that it felt off for an App to fail creation because a ComputeClass references a non-existent/misconfigured PriorityClass. What does feel weird about this is if a PriorityClass gets deleted after a ComputeClass gets created. In that event I added some controller logic on the App to verify that PriorityClass exists.

Reply:
Volume classes are similar. A volume class can reference any storage class it wants; we don't validate that on creation of the volume class.

If the storage class doesn't exist, then the app will fail when it is created because Kubernetes won't be able to find the storage class. It may not return an error, but it will hang forever because the volume won't be created.

Reply:
Okay, I agree with you here then. I've removed the API Server validation. The controller already had logic to check if the PriorityClass exists, do you think that should be removed as well in favor of just letting the Pods create?